stdlib/repr.mli

```diff
@@ -0,0 +1,67 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2025 Kate Deplaix                                          *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Comparison functions.
+
+    @since 5.4 *)
+
+(** {1 Physical comparison} *)
+
+external phys_equal : 'a -> 'a -> bool = "%eq"
+(** [phys_equal e1 e2] tests for physical equality of [e1] and [e2].
+    On mutable types such as references, arrays, byte sequences, records with
+    mutable fields and objects with mutable instance variables,
+    [phys_equal e1 e2] is true if and only if physical modification of [e1]
+    also affects [e2].
+    On non-mutable types, the behavior of [phys_equal] is
+    implementation-dependent; however, it is guaranteed that
+    [phys_equal e1 e2] implies [compare e1 e2 = 0]. *)
+
+(** {1 Polymorphic comparison} *)
+
+external equal : 'a -> 'a -> bool = "%equal"
+(** [equal e1 e2] tests for structural equality of [e1] and [e2].
+    Mutable structures (e.g. references and arrays) are equal
+    if and only if their current contents are structurally equal,
+    even if the two mutable objects are not the same physical object.
+    Equality between functional values raises [Invalid_argument].
+    Equality between cyclic data structures may not terminate.
+    Left-associative operator, see {!Ocaml_operators} for more information. *)
+
+external compare : 'a -> 'a -> int = "%compare"
+(** [compare x y] returns [0] if [x] is equal to [y],
+    a negative integer if [x] is less than [y], and a positive integer
+    if [x] is greater than [y].  The ordering implemented by [compare]
+    is compatible with the comparison predicates [=], [<] and [>]
+    defined above,  with one difference on the treatment of the float value
+    {!Stdlib.nan}.  Namely, the comparison predicates treat [nan]
```

Comment:
“Defined above” is now a dangling pointer, and we probably want to refer to `equal` rather than `(=)`.